### ISSUE TITLE
Legend : need a floating panel with full size legend

### ISSUE THREAD

**Issue by MaelREBOUX** [geosolutions-it/MapStore2#6779]
## Description
Currently the legend of the layers are displayed in the TOC.

But depend of kind of styling it is often unreadeable because very too small :

![image](https://user-images.githubusercontent.com/6370443/114990081-74662180-9e98-11eb-844f-6cf85ecccb27.png)

**What kind of improvement you want to add?** (check one with "x", remove the others)
- [x] Minor changes to existing features
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

I suggest to improve read of the legend by adopting the current way in use in [Mviewer ](https://github.com/geobretagne/mviewer):

![a_better_legend](https://user-images.githubusercontent.com/6370443/114990327-b2fbdc00-9e98-11eb-969b-5beb74ac673a.gif)

**Comment by landryb** [comment]
seconded, also having a single panel with the legendimages from all the visible layers, like it was in mfapp, is something liked by users !

**Comment by LouiseBAILLET** [comment]
Agree with this post. It would be very useful and easier to have a floating legend in Mapstore.

**Comment by catmorales** [comment]
Both proposals are interesting and complementary and go in the direction of a better and quicker understanding of the map, especially as we sometimes have complex legends with many items.
If there is a priority to be given, I am leaning towards implementing Maël's request first.

**Comment by landryb** [comment]
maybe https://mapstore.geo-solutions.it/mapstore/docs/api/plugins#plugins.FloatingLegend is what we need, it's just well hidden somewhere ?

**Comment by tdipisa** [comment]
> maybe https://mapstore.geo-solutions.it/mapstore/docs/api/plugins#plugins.FloatingLegend is what we need, it's just well hidden somewhere ?

@landryb yes, this is more less what you are asking for and it looks similar to the one in description. However, it is a plugin not included in the standard configuration because it is going to be unsupported: no one else has reported any specific interest in it until now :)

**Comment by landryb** [comment]
enabling it locally in my ms2-geor test install, with its default config, here's what it displays:
![image](https://user-images.githubusercontent.com/3421760/115422237-02297000-a1fd-11eb-887f-c12da7dd653d.png)

**Comment by landryb** [comment]
@tdipisa what's the status of the plugin itself within mapstore ? as you say it's unsupported and i see there are no translations for it, was it an old plugin nobody is maintaining ? or something that can be improved/enabled ? i can do the mapstore PR adding translations...

**Comment by tdipisa** [comment]
@landryb this is an old plugin that nobody is maintaining at the moment and we have decided to not include it in the default configuration, also because we found the current implementation not so useful: in my opinion it should be reviewed in some way. There is also a pending discussion in the MapStore team to completely remove this tool in favor of other better solutions (to evaluate), but there isn't a final decision on this for now. Feel free anyway to propose your ideas and to contribute on it if you want, let's discuss maybe about this first. For what concerns the geOrchestra issue anyway: I kindly suggest you to clarify about the future of this plugin before enabling it in geOrchestra.

**Comment by landryb** [comment]
well, i think it has its own usecase, if you want to provide users a way to simply see all layers legends without giving them the fullblown layertree (even if i know, one can disable the sub-plugins/features) - it takes less space than layertree (default group, etc..), and with layertree the layer legends are hidden by default. also, with floatingpanel you only can hide a layer, you cant reorder them - in some cases you might not want users to reorder layers, etc....

but yes, i guess with fine-tuning of layertree, one can more or less get the same feature set. I'll leave the others chime in on this topic :) after all, the original issue was to have a popover (plz, no floating windows!!!) panel with a larger legend.. `floatinglegend` might allow that via `width` setting, tested working here.

**Comment by MaelREBOUX** [comment]
Well. our alpha testers pointed out that access to the legend is so far less usable in MapStore than in Mapfishapp :(
And I can't say they are wrong.

In my point of view we have to discuss **quickly** on that feature (expand the image legend in a kind of popup; like in mviewer)

**Comment by landryb** [comment]
Fwiw, on https://mapstore.geosolutionsgroup.com/mapstore/#/viewer/openlayers/7920 i've seen that one can have a widget with the map legend, @tdipisa is this using the deprecated `FloatingLegend` plugin or should that be the 'right way' of having a floating panel with the legend ?

**Comment by tdipisa** [comment]
> Fwiw, on https://mapstore.geosolutionsgroup.com/mapstore/#/viewer/openlayers/7920 i've seen that one can have a widget with the map legend, @tdipisa is this using the deprecated `FloatingLegend` plugin or should that be the 'right way' of having a floating panel with the legend ?

@landryb that's simply a quick trick you can do with the current text widget, see below.

![legend_widget](https://user-images.githubusercontent.com/1280027/152542199-10488bf4-ed21-4ffb-b85f-b55d05282b91.gif)

**Comment by landryb** [comment]
Hah, right, nice cheat ;) i thought it was a proper legend panel, but it's just a text widget embedding the getlegendgraphics url so that it refreshes ?

**Comment by MaelREBOUX** [comment]
What a trick !

But does not resolve entirely the question.
